File: backend/src/main/java/com/plugga/backend/dao/SideDAO.java
package com.plugga.backend.dao;

import com.plugga.backend.entity.Side;

import java.util.List;

public interface SideDAO {
    List<Side> findAll();

    Side findById(int id);

    List<Side> findByCardId(int cardId);

    void save(Side side);

    void deleteById(int id);
}

File: backend/src/main/kotlin/com/plugga/backend/dao/SideDAO.kt
package com.plugga.backend.dao

import com.plugga.backend.entity.Side
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.repository.PagingAndSortingRepository
import org.springframework.stereotype.Repository

@Repository
interface SideDAO : PagingAndSortingRepository<Side, Int> {

    fun findByCardId(cardId: Int): List<Side>

    fun findByCardId(pageable: Pageable, cardId: Int): Page<Side>
}
